examples/pom_file_generation/BUILD

```diff
@@ -0,0 +1,28 @@
+load("@bazel_common//tools/maven:pom_file.bzl", "pom_file")
+load("@rules_jvm_external//:defs.bzl", "artifact")
+
+pom_file(
+    name = "guice_pom",
+    targets = [
+        artifact("com.google.inject:guice"),
+    ],
+    template_file = "pom_template.xml",
+)
+
+java_binary(
+    name = "my_binary",
+    srcs = ["java/com/example/bazel/Main.java"],
+    main_class = "com.example.bazel",
+    deps = [
+        artifact("com.google.inject:guice"),
+        artifact("com.google.guava:guava"),
+    ],
+)
+
+pom_file(
+    name = "my_binary_pom",
```

Comment:
I think it's confusing that you're creating a pom file for a binary. pom files are usually shipped with libraries

Reply:
Switched to java_library. Thanks!